code/modules/projectiles/guns/projectile.dm

```diff
@@ -124,10 +124,7 @@
 
 /obj/item/weapon/gun/projectile/process_chambered()
 	var/obj/item/ammo_casing/AC = getAC()
-	if(in_chamber)
-		return 1 //{R}
-	if(isnull(AC) || !istype(AC))
-		return
+	can_discharge()
```

Comment:
Yeah, but both of those checks return, and they return different values. So if you just moved those checks to a separate proc and didn't change the return values, you can't do that.